core/src/main/java/com/amplifyframework/auth/AuthException.java

```diff
@@ -51,4 +51,144 @@ public AuthException(
     ) {
         super(message, recoverySuggestion);
     }
+
+    /**
+     * Auth exception caused by the user being signed out.
+     */
+    public static class SignedOutException extends AuthException {
```

Comment:
Ooh are we bringing back category sub-exceptions? I liked the design, but all of our other categories don't have them at the moment right?

Reply:
Yeah we need to discuss this as a team for adding to the other categories but for now, added this because it was super necessary for Auth.